src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs

```diff
@@ -362,9 +383,17 @@ public static byte[] GenerateKeyBytes(int sizeInBits)
                     securityKey = new SymmetricSecurityKey(GenerateKeyBytes(384));
                 else if (SecurityAlgorithms.Aes256CbcHmacSha512.Equals(encryptingCredentials.Enc))
                     securityKey = new SymmetricSecurityKey(GenerateKeyBytes(512));
+
+                // only 128, 192 and 256 AesGcm for CEK algorithm
+                else if(SecurityAlgorithms.Aes128Gcm.Equals(encryptingCredentials.Enc))
+                    securityKey = new SymmetricSecurityKey(GenerateAesGcmKeyBytes(128));
+                else if (SecurityAlgorithms.Aes192Gcm.Equals(encryptingCredentials.Enc))
+                    securityKey = new SymmetricSecurityKey(GenerateAesGcmKeyBytes(192));
+                else if (SecurityAlgorithms.Aes256Gcm.Equals(encryptingCredentials.Enc))
+                    securityKey = new SymmetricSecurityKey(GenerateAesGcmKeyBytes(256));
                 else
                     throw LogHelper.LogExceptionMessage(
-                        new SecurityTokenEncryptionFailedException(LogHelper.FormatInvariant(TokenLogMessages.IDX10617, LogHelper.MarkAsNonPII(SecurityAlgorithms.Aes128CbcHmacSha256), LogHelper.MarkAsNonPII(SecurityAlgorithms.Aes192CbcHmacSha384), LogHelper.MarkAsNonPII(SecurityAlgorithms.Aes256CbcHmacSha512), LogHelper.MarkAsNonPII(encryptingCredentials.Enc))));
```

Comment:
Even if I provide a custom `CryptoProviderFactory` with my own implementation of `AuthenticatedEncryptionProvider`, this code still throws and prevents me from creating tokens of type AESGCM

One alternate approach would be to have two separate PRs:

1. Enable support for AES-GCM with custom CryptoProviderFactory. This would allow consumers of the API to plug-in their own AuthenticatedEncryptionProvider (e.g. Windows Interop / .NET AES GCM) 

2. A separate PR that adds support for AES-GCM Encryption. Later .NET APIs have a cross-platform API to call, but folks on .NET Framework would need to use Windows Interop